src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -1023,18 +1023,21 @@ void eventUpdated(AsyncEvent e) throws ClosedChannelException {
 
         // This returns immediately. So caller not allowed to send/receive
         // on connection.
-        synchronized void register(AsyncEvent e) {
-            if (closed) e.abort(selectorClosedException());
-            registrations.add(e);
-            selector.wakeup();
-        }
-
-        synchronized void cancel(SocketChannel e) {
-            SelectionKey key = e.keyFor(selector);
-            if (key != null) {
-                key.cancel();
+        void register(AsyncEvent e) {
+            var closed = this.closed;
+            if (!closed) {
+                synchronized (this) {
+                    closed = this.closed;
+                    if (!closed) {
+                        registrations.add(e);
+                    }
+                }
+            }
+            if (closed) {
+                e.abort(selectorClosedException());
+            } else {
+                selector.wakeup();
```

Comment:
Hello Daniel, before this PR, except for the `wakeupSelector()` method on `SelectorManager`, all other methods which were operating on the `selector` field were `synchronized` on the `SelectorManager` instance. After this PR, that continues to be true except for this specific line where the operation on the `selector` field is outside of a `synchronized` block. Would that be OK?

Reply:
Thank you for that explanation. I understand this change better now.